zookeeper-server/src/main/java/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -957,7 +962,11 @@ void readResponse(ByteBuffer incomingBuffer) throws IOException {
 
         SendThread(ClientCnxnSocket clientCnxnSocket) {
             super(makeThreadName("-SendThread()"));
-            state = States.CONNECTING;
+            try {
+                changeZkState(States.CONNECTING);
+            } catch (IOException e) {
+                throw new RuntimeException("Connection setup failed when migrate state to CONNECTING");
```

Comment:
Throwing a RuntimeException may lead to unexpected behaviour. 
So we have a way to throw a checked exception and deal explicitly with this case?

Reply:
Thanks you for your comments. I accept your suggestion. I throw checked IOException for SendThread constructor. Please review latest patch.